### ISSUE TITLE
Support for Mac Catalyst

### ISSUE THREAD

**Issue by robertying** [facebook/hermes#460]
## Bug Description

React Native 0.64-rc has introduced Hermes for iOS. I'm trying to build for Mac Catalyst and there seems to be some problem with prebuilt binary shipped with Hermes for iOS.

```
ld: building for Mac Catalyst, but linking in dylib built for iOS Simulator, file '/Users/rying/myapp/ios/Pods/hermes-engine/destroot/Library/Frameworks/iphoneos/hermes.framework/hermes' for architecture x86_64
```

<del>- [ ] I have run `gradle clean` and confirmed this bug does not occur with JSC</del>

Hermes version: 0.7.2
React Native version (if any): 0.64.0-rc.3
OS version (if any): macOS 11.2.1
Platform (most likely one of arm64-v8a, armeabi-v7a, x86, x86_64): x86_64 (Catalyst)

## Steps To Reproduce

1. Init RN with 0.64.0-rc.3
2. Build for Mac Catalyst target

## The Expected Behavior

I see Hermes already supports x86-64 so it should play nice with Catalyst too. 

Thanks for the great work and I really hope it gets fixed soon!

**Comment by Huxpro** [comment]
@robertying Hey thanks for reporting this interesting use case! Just let you know that the community is currently prioritizing shipping the stable 0.64 first before investigating additional features. cc @grabbou

**Comment by Arkkeeper** [comment]
This issue seems to be related with mine #468 

Hermes as a framework is compiled for certain architectures and platforms (iOS, iOS Simulator and macOS), and then they are merged into a fat framework with lipo.

For a proper compatibility with Mac Catalyst (and arm64 iOS Simulators on M1 Macs) Hermes should be compiled as .xcframework containing slices for all necessary archs ans platforms. For example, OpenSSL.xcframework shipped along with Flipper, has 4 slices:

<img width="321" alt="openssl" src="https://user-images.githubusercontent.com/20115419/111457155-e6780900-8728-11eb-9b98-6d1eab6dc4c2.png">

I've forked Hermes and managed to add arm64 iOS Simulator support. Unfortunately, I haven't figure yet how we should pass necessary cmake flags for building Catalyst.

Our team has been using using React Native app with Catalyst in production for more than a year, and Hermes support would be great.

**Comment by Huxpro** [comment]
This is fixed by https://github.com/facebook/hermes/pull/546 (was #543 and #475) and should be available when [Hermes 0.8.1](https://github.com/facebook/hermes/issues/528) is out (together with RN 0.65 stable).
